composer/trainer/trainer.py

```diff
@@ -475,13 +477,20 @@ def device(self):
         # No setter for device since it wouldn't make sense to change it
         return self._device
 
+    @device.setter
+    def device(self, device: None):
+        del device  # unused
+        raise AttributeError("To change the training device, instead create a new Trainer.")
+
     @property
     def grad_accum(self):
         """Gradient Accumulation"""
         return self.state.grad_accum
 
     @grad_accum.setter
     def grad_accum(self, grad_accum: int):
+        if self.deepspeed_enabled:
+            raise RuntimeError("Cannot change grad_accum when using deepspeed")
```

Comment:
Could you say what should be done instead? Is this another "If you'd like to change it, create a new Trainer" type of thing?